### ISSUE TITLE
At least one Döner place (Yufka's in Heidelberg) is not shown in doenieren.in, despite being in OSM

### ISSUE THREAD

**Issue by TuringTux** [midzer/doenieren#4]
Yufka's Kebap & Pizza in Heidelberg (https://www.openstreetmap.org/node/4346900397) is not shown in https://doenieren.in/heidelberg/, despite being entered into OSM.

The `cuisine` of the particular node is `kebab;pizza` – however if I read #3 correctly, this shouldn't be a problem, as you took care for that in https://github.com/midzer/pbf2md/commit/1728ba2d06c75e6bdcc2f2f93cb212865e5e4a83.

I'd be happy to help fixing that bug. If you need any further information, just ping me :)

**Comment by midzer** [comment]
Hi,

thanks for opening up :)

I figured out mentioned node was missing `addr:city`, which is necessary for preprocessing.

Have a nice week
midzer

**Comment by TuringTux** [comment]
I see, thanks for editing!

This is likely out of scope, but I fear that this problem (nodes lacking `addr:city`) might be more common and could cause more Döner places to not appear on the map.

Is this something that doenieren.in could deal with, e.g. by checking that the coordinates are within city boundaries? (sounds hard)

Or is it maybe something that needs a more concerted effort from the OSM side? I suppose someone with GIS experience could write queries to find all OSM nodes without `addr:city` within a city and than batch-add them in...

**Comment by midzer** [comment]
> This is likely out of scope, but I fear that this problem (nodes lacking addr:city) might be more common and could cause more Döner places to not appear on the map.

Yeah, thats a pity.

> Is this something that doenieren.in could deal with, e.g. by checking that the coordinates are within city boundaries? (sounds hard)

I stumbled upon the problem (missing `addr:city`) already for shoogle project https://github.com/lokal-ninja
From mathematical point of view, calculating whether a coordinate is within city boundaries is possible. But where to get each cities boundaries coordinates to use them in preprocessing https://github.com/midzer/pbf2md/tree/doener ?

> Or is it maybe something that needs a more concerted effort from the OSM side? I suppose someone with GIS experience could write queries to find all OSM nodes without addr:city within a city and than batch-add them in...

Yeah, that would be awesome. Consistent data is always good. I'm not so involved in the OSM community at the moment. Might be a good time to re-join IRC :)

**Comment by midzer** [comment]
Should be doable with something like https://osm2pgsql.org/examples/road-length/

**Comment by TuringTux** [comment]
Yes, that sounds possible. Probably a nice project if one had the time.
